go/vt/vtgate/executor.go

```diff
@@ -195,27 +197,172 @@ func (e *Executor) Execute(ctx context.Context, method string, safeSession *Safe
 	return result, err
 }
 
-func saveSessionStats(safeSession *SafeSession, stmtType sqlparser.StatementType, result *sqltypes.Result, err error) {
+// StreamExecute executes a streaming query.
+func (e *Executor) StreamExecute(
+	ctx context.Context,
+	method string,
+	safeSession *SafeSession,
+	sql string,
+	bindVars map[string]*querypb.BindVariable,
+	callback func(*sqltypes.Result) error,
+) error {
+	span, ctx := trace.NewSpan(ctx, "executor.StreamExecute")
+	span.Annotate("method", method)
+	trace.AnnotateSQL(span, sql)
+	defer span.Finish()
+
+	logStats := NewLogStats(ctx, method, sql, bindVars)
+
+	var err error
+	var stmtType sqlparser.StatementType
+	var rowsAffected sync2.AtomicInt64
+	var rowsReturned sync2.AtomicInt64
+	var insertID sync2.AtomicInt64
+	err = e.newExecute(ctx, safeSession, sql, bindVars, logStats, func(plan *engine.Plan, vc *vcursorImpl, bindVars map[string]*querypb.BindVariable, execStart time.Time) error {
```

Comment:
small nit, for code readability it would be nice to export the first function we are passing to `e.newExecute`. that function is pretty long.

Reply:
it is refactored now